packages/core/src/metadata/ReflectMetadataProvider.ts

```diff
@@ -20,14 +20,27 @@ export class ReflectMetadataProvider extends MetadataProvider {
     // as we were mapping it to UnknownType which is a string. This is to prevent defaulting to JSON
     // column type, which can be often hard to revert and cause hard to understand issues with PKs.
     if (prop.kind === ReferenceKind.SCALAR && type === Object) {
-      type = String;
+      this.getTypeFromDefault(prop);
     }
 
-    prop.type = type.name;
+    prop.type = type?.name;
 
     if (prop.type && ['string', 'number', 'boolean', 'array', 'object'].includes(prop.type.toLowerCase())) {
       prop.type = prop.type.toLowerCase();
     }
   }
 
+  private getTypeFromDefault(prop: EntityProperty): any {
+    switch (typeof prop.default) {
+      case 'number':
+        return Number;
+      case 'boolean':
+        return Boolean;
+      case 'object':
+      case 'string':
+      default:
+        return String;
+    }
+  }
```

Comment:
this can be greatly simplified and we should be smarter here, e.g. support at least buffers and dates - there is `UtilsgetObjectType()` you can use to get the type from such objects

```suggestion
  private getWrapperTypeObject(value: unknown): string {
    const type = Utils.getObjectType(value);
    return type === 'object' ? 'string' : type;
  }
```

please add more tests for this (can be just more properties in the test you have